src/descriptors/analysis.rs

```diff
@@ -31,6 +31,7 @@ pub enum LianaPolicyError {
     /// The spending policy is not a valid Miniscript policy: it may for instance be malleable, or
     /// overflow some limit.
     InvalidPolicy(miniscript::Error),
+    HardenedMultipath,
```

Comment:
Would it be better to call it `HardenedDerivationStep` (comparing with the comments in the function)?